src/Utilities/Compiler/Extensions/ITypeSymbolExtensions.cs

```diff
@@ -142,10 +142,11 @@ public static bool DerivesFrom([NotNullWhen(returnValue: true)] this ITypeSymbol
             INamedTypeSymbol? iDisposable,
             INamedTypeSymbol? iAsyncDisposable)
         {
-            if (type.IsReferenceType)
+            // structs can be Disposable as well
+            if (type.IsReferenceType || type.IsValueType)
             {
-                return IsInterfaceOrImplementsInterface(type, iDisposable)
-                    || IsInterfaceOrImplementsInterface(type, iAsyncDisposable);
+                if (IsInterfaceOrImplementsInterface(type, iDisposable) || IsInterfaceOrImplementsInterface(type, iAsyncDisposable))
```

Comment:
Could you please revert that part of the change as it's not needed?

Reply:
If the type is a ValueType and doesn't implement IDisposable we still need to check if it's a ref struct that implements the Dispose shape. Wouldn't reverting this change return without checking ref structs?